ambry-store/src/main/java/com/github/ambry/store/Journal.java

```diff
@@ -117,6 +117,24 @@ void addEntry(Offset offset, StoreKey key, Long crc) {
     }
   }
 
+  /**
+   * This method should only be called when replica gets sealed. Remove all entries belongs to the auto closed log segment.
+   * It will delay the addEntry method for sealed replica, so the replication performance will be hold while running this method.
+   * @param activeLogSegmentName journal should remove all entries not belongs to active log segment.
+   */
+  synchronized void cleanUpJournal(LogSegmentName activeLogSegmentName) {
+    while (!inBootstrapMode) {
```

Comment:
If `inBootstrapMode` is False, the loop will eat a lot of CPU. Can we do sleep wait?

Reply:
Actually I have other condition to break the loop. To be more clear, I move that condition inside the while loop check, also, I'm not sure for the performance which way is better. 1. each time when we remove the entry, we release the lock.
2. release the lock wait until the cleanup journal finished. Since it's for sealed replica only, I'm planning to use the config to switch between the two approach test the performance.